lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/storage/AzureStorageUtil.java

```diff
@@ -0,0 +1,150 @@
+    final CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+    try {
+      final CloudBlobContainer container = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
+      return container.getDirectoryReference(jobFolder).getUri();
+    } catch (URISyntaxException | StorageException e) {
+      throw new IOException("Failed to get the job submission folder URI", e);
+    }
+  }
+
+
+  public String createContainerSharedAccessSignature() throws IOException {
+    try {
+      CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+      CloudBlobContainer cloudBlobContainer = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
+      cloudBlobContainer.createIfNotExists();
+
+      return cloudBlobContainer.generateSharedAccessSignature(getSharedAccessContainerPolicy(), null);
+
+    } catch (StorageException | URISyntaxException | InvalidKeyException e) {
+      throw new IOException("Failed to generate a shared access signature for storage container.", e);
+    }
+  }
+
+
+  /**
+   * Upload a file to the storage account.
+   *
+   * @param jobFolder the path to the destination folder within storage container.
+   * @param file the source file.
+   * @return the SAS URI to the uploaded file.
+   * @throws IOException
+   */
+  public URI uploadFile(final String jobFolder, final File file) throws IOException {
+
+    LOG.log(Level.INFO, "Uploading [{0}] to [{1}]", new Object[]{file, jobFolder});
+
+    try {
+      final CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+      final CloudBlobContainer container = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
+
+      final String destination = String.format("%s/%s", jobFolder, file.getName());
+      final CloudBlockBlob blob = container.getBlockBlobReference(destination);
+
+      try (FileInputStream fis = new FileInputStream(file)) {
+        blob.upload(fis, file.length());
+      }
+
+      LOG.log(Level.FINE, "Uploaded to: {0}", blob.getStorageUri().getPrimaryUri());
+      return this.cloudBlobClientProvider.generateSharedAccessSignature(blob, getSharedAccessBlobPolicy());
+
+    } catch (final URISyntaxException | StorageException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private SharedAccessBlobPolicy getSharedAccessBlobPolicy() {
+
+    Date now = Calendar.getInstance().getTime();
+
+    Calendar calendar = Calendar.getInstance();
+    calendar.add(Calendar.MINUTE, BLOB_SAS_TOKEN_VALIDITY_MINUTES);
+    Date tokenExpirationDate = calendar.getTime();
+
+    final SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
+    policy.setPermissions(EnumSet.of(SharedAccessBlobPermissions.READ));
+    policy.setSharedAccessStartTime(now);
+    policy.setSharedAccessExpiryTime(tokenExpirationDate);
+
+    return policy;
+  }
+
+  private SharedAccessBlobPolicy getSharedAccessContainerPolicy() {
```

Comment:
nit: Pretty much identical to getSharedAccessBlobPolicy. Consider creating a shared method.

Reply:
changed.